TombEngine/Game/Lara/lara_helpers.cpp

```diff
@@ -68,28 +69,199 @@ void HandleLaraMovementParameters(ItemInfo* item, CollisionInfo* coll)
+	while (nextIndex != *currentIndex)
+	{
+		auto nextWeaponType = SCROLL_WEAPON_TYPES[nextIndex];
+		if (player.Weapons[(int)nextWeaponType].Present)
+			return nextWeaponType;
+
+		nextIndex = getNextIndex(nextIndex);
+	}
+
+	// No valid weapon type; return nullopt.
+	return std::nullopt;
+}
+
+void HandlePlayerQuickActions(ItemInfo& item)
+{
+	auto& player = GetLaraInfo(item);
+
+	// Handle medipacks.
+	if (IsClicked(In::SmallMedipack) || IsClicked(In::LargeMedipack))
+		UsePlayerMedipack(item);
+
+	// Handle weapon scroll requests.
+	if (IsClicked(In::PreviousWeapon) || IsClicked(In::NextWeapon))
+	{
+		bool getPrev = IsClicked(In::PreviousWeapon);
+		auto weaponType = GetPlayerScrolledWeaponType(item, player.Control.Weapon.GunType, getPrev);
+
+		if (weaponType.has_value())
+			player.Control.Weapon.RequestGunType = *weaponType;
+	}
+
+	// Handle weapon requests.
+	if (IsClicked(In::Weapon1) && player.Weapons[(int)LaraWeaponType::Pistol].Present)
+		player.Control.Weapon.RequestGunType = LaraWeaponType::Pistol;
+
+	if (IsClicked(In::Weapon2) && player.Weapons[(int)LaraWeaponType::Shotgun].Present)
+		player.Control.Weapon.RequestGunType = LaraWeaponType::Shotgun;
+
+	if (IsClicked(In::Weapon3) && player.Weapons[(int)LaraWeaponType::Uzi].Present)
+		player.Control.Weapon.RequestGunType = LaraWeaponType::Uzi;
+
+	if (IsClicked(In::Weapon4) && player.Weapons[(int)LaraWeaponType::Revolver].Present)
+		player.Control.Weapon.RequestGunType = LaraWeaponType::Revolver;
+
+	if (IsClicked(In::Weapon5) && player.Weapons[(int)LaraWeaponType::GrenadeLauncher].Present)
+		player.Control.Weapon.RequestGunType = LaraWeaponType::GrenadeLauncher;
+
+	if (IsClicked(In::Weapon6) && player.Weapons[(int)LaraWeaponType::Crossbow].Present)
+		player.Control.Weapon.RequestGunType = LaraWeaponType::Crossbow;
+
+	if (IsClicked(In::Weapon7) && player.Weapons[(int)LaraWeaponType::HarpoonGun].Present)
+		player.Control.Weapon.RequestGunType = LaraWeaponType::HarpoonGun;
+
+	if (IsClicked(In::Weapon8) && player.Weapons[(int)LaraWeaponType::HK].Present)
+		player.Control.Weapon.RequestGunType = LaraWeaponType::HK;
+
+	if (IsClicked(In::Weapon9) && player.Weapons[(int)LaraWeaponType::RocketLauncher].Present)
+		player.Control.Weapon.RequestGunType = LaraWeaponType::RocketLauncher;
+
+	// TODO: 10th possible weapon, probably grapple gun.
+	/*if (IsClicked(In::Weapon10) && player.Weapons[(int)LaraWeaponType::].Present)
+	player.Control.Weapon.RequestGunType = LaraWeaponType::;*/
+
+	// TODO: Could theoretically remove SwitchTarget and instead do unique behaviour handling using only Look. -- Sezz 2023.07.08
+	// HACK: Handle target switch when locked on to an entity.
+	if (player.Control.HandStatus == HandStatus::WeaponReady &&
+		player.TargetEntity != nullptr)
+	{
+		if (IsClicked(In::Look))
+		{
+			ActionMap[(int)In::SwitchTarget].Update(true);
+			//ActionMap[(int)In::Look].Clear();
+		}
+		else
+		{
+			ClearAction(In::SwitchTarget);
```

Comment:
I don't fully understand the difference between `In::Look` and `In::SwitchTarget` in this context.